src/hooks/useRoom.ts

```diff
@@ -22,12 +21,11 @@ export const useRoom = ({ room, venueName }: UseRoomProps) => {
 
   const roomUrl = room?.url ?? "";
 
-  // @debt This selector relies on all venues in firebase being loaded into memory.. not very efficient
-  const venues = useSelector(orderedVenuesSelector);
+  const { relatedVenues } = useRelatedVenues({});
```

Comment:
This seems wrong? I suspect it was done because you have no need for 'current venue', yet the props are required (even though `currentVenueId` is optional)?

---

[potentially-OOS] I was going to suggest that we should just pass in the `venue.id` since it was in-scope anyway, but it seems like we end up finding the venue based on this data, so that wouldn't work.

I was also going to suggest that we should potentially refactor `useRoom` to accept `venueId` rather than `venueName`, but it's used `~5` locations across the app currently, and at least 2 of them don't immediately have access to the `venue` directly (only `venueName`) which could complicate this further.. though looking at those cases:

- `EventModal`: `src/components/organisms/EventModal/EventModal.tsx`
  - This seems to have `event.venueId` though, which is all we would need.. and this is already passed to `useRelatedVenues` to get `eventVenue`, so this is fine.
- `RoomModalContent`: `src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx`
  - This is only rendered by `RoomModal`, which already has `venue` present, so it could just pass on the `venue.id` rather than `venue.name`.

So it seems like it wouldn't be all that hard to do, and would make `useRoom` more canonical to our standards, while also avoiding this weird hacky looking syntax here at the moment.

If you're going to call OOS, then at least add 2 `@debt` comments, one to refactor `useRoom` to take `venueId` instead of `venueName` (at the top of this hook), and one on this line to say to pass `venueId` through to `useRelatedVenues` when we do; and then also please create a backlog issue that references this snippet + details for future.

---

[OOS] It would be a tiny change to make the props optional so we don't need to hack around it like this.. but that would seemingly require changing `ReactHook` + ensuring everywhere that uses that would work properly in that case.. so maybe best to leave that to a future problem.

Reply:
Added `@debt` comments in 9b83f5d27a1ee3495619618af0a00b9f28337ab7 and opened a stub of an issue https://github.com/sparkletown/internal-sparkle-issues/issues/911